crypto/armcap.c

```diff
@@ -71,6 +72,20 @@ void OPENSSL_cpuid_setup(void) __attribute__ ((constructor));
 #   define OSSL_IMPLEMENT_GETAUXVAL
 #  endif
 # endif
+# if defined(__FreeBSD__) && __FreeBSD_version >= 1200000
```

Comment:
It looks like elf_aux_info() has been a public function on stable/11 since r356674, so 1103507 might be a better check.
Also, ISTR a need to use separate preprocessor directives for testing if a symbol is defined and comparing against its value, since some platforms we claim to support will choke on expressions like this if the value is not defined.

Reply:
Well the man page specifies since the 12.x release https://github.com/freebsd/freebsd/blob/b5a8a00c4307f6a1024e85e779c8e418ee087127/lib/libc/gen/auxv.3#L98

indeed even the last 11 release, the 11.4 one does not even have this man page as seen here https://github.com/freebsd/freebsd/blob/release/11.4.0/lib/libc/gen/auxv.3 so it seems "wiser" approach.

Reply:
> Also, ISTR a need to use separate preprocessor directives for testing if a symbol is defined and comparing against its value, since some platforms we claim to support will choke on expressions like this if the value is not defined.

I do not get the last part, it is very common to test if this FreeBSD then its release via the __FreeBSD_version constant.

Reply:
It is very common to do that *on FreeBSD*, yes.  But this file gets compiled on all OSes that OpenSSL supports (well, all those that support arm hardware), and some of those compilers leave much to be desired.